### ISSUE TITLE
Implement `--remote-wait`

### ISSUE THREAD

**Issue by clason** [neovim/neovim#24788]
### Problem

`:h --remote-wait` says "Not yet supported by Nvim".

### Expected behavior

`--remote-wait` is supported by Nvim.

(Can't use `neovim --remote` as git editor, since the invocation immediately terminates after launching the remote instance, yielding "Aborting commit due to empty commit message.")

**Comment by justinmk** [comment]
My proposal in https://github.com/neovim/neovim/pull/18414 is that `--remote` always waits (blocks until return). And if one wants asynchronous (immediate return) behavior, prepending `:&` to a command opts-into that behavior.

**Comment by gitmpr** [comment]
Neovim 0.12 has shipped — which was the prerequisite @justinmk mentioned before considering eager-merge of #37951.

That PR implemented exactly this (blocking `--remote` by default, `+async` opt-out) and received a positive review ("the right shape"), but was auto-closed by the bot after the contributor went quiet during the rebase cycle. Nothing was wrong with the approach.

The practical motivation for getting this done: `mhinz/neovim-remote` (`nvr`) has had no upstream commits since May 2022 and is effectively abandoned. Its `--remote-wait` semantics — blocking the calling process until the buffer is unloaded, needed for `$EDITOR` inside `:terminal` — is the only remaining feature native Neovim doesn't cover. Once this lands, `nvr` has no reason to exist.

Would a new PR continuing from #37951's design be welcome? Happy to take it on.

**Comment by justinmk** [comment]
I'm still planning to revisit #37951, but an active and responsive revival would be welcome. It helps a lot if the contributor thinks carefully about the work and keeps the PR rebase, and tries to anticipate edge cases and think about the design. Otherwise it makes review very slow. @rafaelcl292 are you able to keep maintaining the PR or would you prefer someone to take it over?

**Comment by rafaelcl292** [comment]
@justinmk just rebased that branch. feel free to reopen #37951. I’ll do my best to help get it landed.
